src/Symfony/Component/Serializer/Normalizer/ConstraintViolationListNormalizer.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\Validator\ConstraintViolationListInterface;
+
+/**
+ * A normalizer that uses an objects own JsonSerializable implementation.
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class ConstraintViolationListNormalizer implements NormalizerInterface
+{
+    public function normalize($violationsList, $format = null, array $context = array())
+    {
+        $violations = array();
+        $messages = array();
+
+        foreach ($violationsList as $violation) {
+            $propertyPath = $violation->getPropertyPath();
+
+            $violations[] = array(
+                'propertyPath' => $propertyPath,
+                'message' => $violation->getMessage(),
+            );
+
+            $prefix = $propertyPath ? sprintf('%s: ', $propertyPath) : '';
+
+            $messages[] = $prefix.$violation->getMessage();
+        }
+
+        return array(
+            'title' => 'An error occurred',
```

Comment:
Isnt this format rather opinionated? Do we really need it compared to `return $violations;`.

Point is; i like to integrate such a normalizer in one of my api response providers, by doing something like;

```
return new JsonResponse(['our_api_format' => ['errors' => $normalizer->normalize($violations)]]);
```

Any thoughts?